common/go-utils/src/inv.rs

```diff
@@ -111,35 +61,55 @@ impl Inventory {
     }
 }
 
-#[derive(Debug, Deserialize, Serialize, Clone)]
-struct Tag {
-    #[serde(alias = "ref")]
-    reference: String,
+#[derive(Debug, Deserialize)]
+struct GoRelease {
+    files: Vec<GoFile>,
+}
+
+#[derive(Debug, Deserialize)]
+struct GoFile {
+    os: String,
+    arch: String,
+    sha256: String,
+    version: String,
+}
+
+impl GoFile {
+    fn target_arch(&self) -> String {
+        format!("{}-{}", self.os, self.arch)
```

Comment:
Good point -- I've refactored target_arch to return an Option<string> value instead in https://github.com/heroku/buildpacks-go/pull/216/commits/a7035c5038d1da56cd3ff0e344443107061b47b7 for now. Does this seem more palatable? :)

The plan is to move away from using "target_arch" at all, and instead set it as `os` and `arch` fields on the Artifacts stored in our inventory. As part of that change I also plan to:

* Remove the Artifact#architecture field (which currently holds the value returned by the GoFile#target_arch function).
* Filter artifacts to store in our inventory based on the `os` and `arch` fields (rather than the concatenated value).

In the interest of keeping changes in this PR minimal I'd suggest I make those changes in a follow-up PR. This PR currently reflects the changes to `inventory.toml` with the current Artifact fields. Removing `architecture` and settings `os` and `arch` instead would add a lot of noise as every entry will be updated.

Does that sound like a good approach?